package.json

```diff
@@ -7,10 +7,11 @@
   },
   "scripts": {
     "test": "npm run lint && npm run nyc && npm run e2e",
-    "nyc": "nyc mocha --recursive -t 10000",
-    "e2e": "node ./scripts/test.js",
     "lint": "npx eslint .",
     "list": "node ./scripts/list.js",
+    "nyc": "nyc mocha --recursive -t 10000",
+    "e2e": "node ./scripts/main.js",
```

Comment:
`e2e` should call the `test.js` script, the Travis build uses it.

Reply:
@aklenik I've actually removed `test.js` in favour of `main.js` which comprises similar functionality.

Reply:
@gregdhill Similar, but not the same :) During the Travis CI, the `.travis/script.sh` script will be called with the `BENCHMARK` env. variable set to different values. This value is passed to `test.js` as the only argument. Because of this, the current call to `main.js` will fail: 
```
> caliper@0.1.0 e2e /home/klenik/projects/prs/start-script
> node ./scripts/main.js "simple"

error: [scripts/main.js]: config file is required
```

You're right, that `main.js` should be used, it's cleaner and doesn't assume defaults that need to be maintained. But in this case you should also modify `.travis/script.sh` at lines 27, 35 and 39.

Line 27: 
```
npm run test -- -- --config=benchmark/composer/config.yaml --network=network/fabric-v1.2/dev/composer-tls.json
```

Line 34-35:
```
npm run fabric-v1.1-deps
npm run test -- -- --config=benchmark/drm/config.yaml --network=network/fabric-v1.1/dev/fabric-go-tls.json
```

Line 39:
```
npm run test -- -- --config=benchmark/simple/config-linear-rate.yaml --network=network/fabric-v1.1/dev/fabric-node-tls.json
```

__(Note the double `-- --`, since the parameters pass through two run calls.)__

NB: the CI process needs to be improved, that's why I suggested not to bother with `test.js`, but your way works too, and already starts the CI improvements :)

Reply:
Ahh ofc, I've made your suggested changes! :)